### PR TITLE
Improving labeling of side inputs for Dataflow

### PR DESCRIPTION
**Pull request by pabloem**
Side inputs in Python show the internal step name on their user-visible label. This is not useful, and in fact undesirable as it also hides what kind of side input the pcollection is.
This PR improves the labeling of side inputs by showing the side input kind and pcollection in the label.

r: @robertwb 
I have three questions regarding this change:
* A side input that is passed twice to the same ParDo may have problems here. But this may be a scenario that we want to discourage anyway. What do you think?
* I feel that the `'/Do' ` relabeling of the ParDo in line 525 could also be reconsidered to something more intuitive for the users (we've had bugs filed because users can't figure out which step owns their counters).
* Also, the `side_pval.pvalue.producer.full_label.split('/')[-1]` operation I'm adding in line 507 seems a bit sketchy, but I believe it should work in every scenario..

### CODE REVIEW

**Comment by robertwb** [comment]
We definitely want to support the same side input being passed multiple times. Also, I'm all for improving the Do relabeling (though ideally counters would simply be rolled up so one wouldn't need to worry about exact, internal structures). 

The split('/')[-1] should be just fine.

**Comment by pabloem** [comment]
We may want a single `AsSideInput` instance to be passed to multiple different transforms - and this will be no problem. But what about the following:

    pcoll  = 'create_si' >> Create([1, 2, 3, 4, 5])
    pcv = AsIter(pcoll)
    result =  Create([1,2,3]) | beam.Map(lambda x, it1, it2: some_function, it1=pcv, it2=pcv)
    # Passed twice into same ParDo

With my current change, we'd have two transforms with the same `full_label` (e.g. `'AsIter(create_si.out)'`), and that may cause trouble with the UI. Should I make sure to support this case?

**Comment by robertwb** [comment]
Yes, we should make sure to support this case. It wouldn't be hard to add a
counter in case a side input is used more than once.

**Comment by pabloem** [comment]
I've updated the PR with this change. PTAL, and Thanks @robertwb

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 1]
```diff
@@ -497,6 +498,7 @@ def run_ParDo(self, transform_node):
     si_dict = {}
     # We must call self._cache.get_pvalue exactly once due to refcounting.
     si_labels = {}
+    full_label_counts = defaultdict(lambda : 0)
```
Comment (robertwb) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: defaultdict(int)

**Pull request closed without merging**